src/main/java/de/tum/in/www1/artemis/service/iris/IrisSessionService.java

```diff
@@ -55,6 +55,18 @@ public void checkHasAccessToIrisSession(IrisSession session, User user) {
         }
     }
 
+    /**
+     * Checks if the programming exercise for which an Iris operation was requested has Iris activated.
+     * An Iris operation can be performed if the programming exercise has Iris activated.
+     *
+     * @param programmingExercise The programming exercise to check
+     */
+    public void checkIrisActivated(ProgrammingExercise programmingExercise) {
+        if (programmingExercise.getIrisActivated() == null || Boolean.FALSE.equals(programmingExercise.getIrisActivated())) {
+            throw new AccessForbiddenException("Iris not activated for Programming Exercise: " + programmingExercise.getId());
```

Comment:
I don't think this is the correct exception here. `AccessForbiddenException` represents a 403 HTTP error, which is used if the user has e.g. insufficient rights to a resource.
A BadRequest might make more sense.